wasmtime-wasi/src/host.rs

```diff
@@ -4,3 +4,140 @@ include!(concat!(env!("OUT_DIR"), "/wasmtime_ssp.rs"));
+        libc::EILSEQ => wasm32::__WASI_EILSEQ,
+        libc::EINPROGRESS => wasm32::__WASI_EINPROGRESS,
+        libc::EINTR => wasm32::__WASI_EINTR,
+        libc::EINVAL => wasm32::__WASI_EINVAL,
+        libc::EIO => wasm32::__WASI_EIO,
+        libc::EISCONN => wasm32::__WASI_EISCONN,
+        libc::EISDIR => wasm32::__WASI_EISDIR,
+        libc::ELOOP => wasm32::__WASI_ELOOP,
+        libc::EMFILE => wasm32::__WASI_EMFILE,
+        libc::EMLINK => wasm32::__WASI_EMLINK,
+        libc::EMSGSIZE => wasm32::__WASI_EMSGSIZE,
+        libc::EMULTIHOP => wasm32::__WASI_EMULTIHOP,
+        libc::ENAMETOOLONG => wasm32::__WASI_ENAMETOOLONG,
+        libc::ENETDOWN => wasm32::__WASI_ENETDOWN,
+        libc::ENETRESET => wasm32::__WASI_ENETRESET,
+        libc::ENETUNREACH => wasm32::__WASI_ENETUNREACH,
+        libc::ENFILE => wasm32::__WASI_ENFILE,
+        libc::ENOBUFS => wasm32::__WASI_ENOBUFS,
+        libc::ENODEV => wasm32::__WASI_ENODEV,
+        libc::ENOENT => wasm32::__WASI_ENOENT,
+        libc::ENOEXEC => wasm32::__WASI_ENOEXEC,
+        libc::ENOLCK => wasm32::__WASI_ENOLCK,
+        libc::ENOLINK => wasm32::__WASI_ENOLINK,
+        libc::ENOMEM => wasm32::__WASI_ENOMEM,
+        libc::ENOMSG => wasm32::__WASI_ENOMSG,
+        libc::ENOPROTOOPT => wasm32::__WASI_ENOPROTOOPT,
+        libc::ENOSPC => wasm32::__WASI_ENOSPC,
+        libc::ENOSYS => wasm32::__WASI_ENOSYS,
+        // TODO: verify if this is correct
+        #[cfg(target_os = "freebsd")]
+        libc::ENOTCAPABLE => wasm32::__WASI_ENOTCAPABLE,
+        libc::ENOTCONN => wasm32::__WASI_ENOTCONN,
+        libc::ENOTDIR => wasm32::__WASI_ENOTDIR,
+        libc::ENOTEMPTY => wasm32::__WASI_ENOTEMPTY,
+        libc::ENOTRECOVERABLE => wasm32::__WASI_ENOTRECOVERABLE,
+        libc::ENOTSOCK => wasm32::__WASI_ENOTSOCK,
+        libc::ENOTSUP => wasm32::__WASI_ENOTSUP,
+        libc::ENOTTY => wasm32::__WASI_ENOTTY,
+        libc::ENXIO => wasm32::__WASI_ENXIO,
+        libc::EOVERFLOW => wasm32::__WASI_EOVERFLOW,
+        libc::EOWNERDEAD => wasm32::__WASI_EOWNERDEAD,
+        libc::EPERM => wasm32::__WASI_EPERM,
+        libc::EPIPE => wasm32::__WASI_EPIPE,
+        libc::EPROTO => wasm32::__WASI_EPROTO,
+        libc::EPROTONOSUPPORT => wasm32::__WASI_EPROTONOSUPPORT,
+        libc::EPROTOTYPE => wasm32::__WASI_EPROTOTYPE,
+        libc::ERANGE => wasm32::__WASI_ERANGE,
+        libc::EROFS => wasm32::__WASI_EROFS,
+        libc::ESPIPE => wasm32::__WASI_ESPIPE,
+        libc::ESRCH => wasm32::__WASI_ESRCH,
+        libc::ESTALE => wasm32::__WASI_ESTALE,
+        libc::ETIMEDOUT => wasm32::__WASI_ETIMEDOUT,
+        libc::ETXTBSY => wasm32::__WASI_ETXTBSY,
+        libc::EXDEV => wasm32::__WASI_EXDEV,
+        // TODO: add missing defines
+        // #if EOPNOTSUPP != ENOTSUP
+        //      [EOPNOTSUPP] = __WASI_ENOTSUP
+        // #endif
+        // #if EWOULDBLOCK != EAGAIN
+        //      [EWOULDBLOCK] = __WASI_EAGAIN
+        // #endif
```

Comment:
Not sure how to handle these two cases here. Suggestions more than welcome!

Reply:
maybe this:
```rust
#[allow(unreachable_patterns)]    
match errno {
    ...
    libc::EAGAIN | libc::EWOULDBLOCK => __WASI_EAGAIN,
    ...
}
```

Reply:
Perfect, thanks!

Reply:
I've changed it to the way you suggested. I think it looks great now. Have a look and lemme know what you reckon!